### PR TITLE
NPM installer should also work with linux/mac

### PR DESCRIPTION
**Pull request by renanccastro**
In this PR I change a little the way we were doing before, which was using NPM to link the binary in a "path" included directory.
This won't work for multiplatform, because package.json only allows us to specify one possible file for the bin key.

The strategy is discovering which system we are doing the install, and adding .meteor to the execution PATH, windows included.

## How to test?

- Remove or rename your ~/.meteor folder to ~/.meteor-backup (`mv ~/.meteor ~/.meteor-backup`)
- Verify that the npm "meteor" package is not installed, if it is, uninstall it
- Run `npm link` in this package folder

## Docs

- [X] Update Docs - https://github.com/meteor/docs/pull/748
- [X] Update Guide - https://github.com/meteor/guide/pull/1083
- [X] Update tutorials - https://github.com/meteor/react-tutorial/pull/37 + https://github.com/meteor/blaze-tutorial/pull/27 + https://github.com/meteor/svelte-tutorial/pull/10 + https://github.com/meteor/vue-tutorial/pull/11 + https://github.com/meteor/angular-tutorial/pull/1
- [X] Update Website - **Pending publication.**

### CODE REVIEW

File: npm-packages/meteor-installer/install.js [hunk 1]
```diff
@@ -155,16 +176,31 @@ async function extract() {
...
   console.log(`=> Meteor Extracted ${(end - start) / 1000}s`);
+  await setup();
+}
+async function setup(){
   fs.unlinkSync(startedPath);
+  await setupExecPath()
   showGettingStarted();
 }
+async function setupExecPath(){
+  if(os.platform() === 'darwin' || os.platform() === 'linux') {
```
Comment (filipenevola) [inline npm-packages/meteor-installer/install.js:187]: same from above.

```js
if (isWindows) {
  child_process.execSync(`setx path "${meteorPath}/;%path%`);
  return;
}

// non-windows code

File: npm-packages/meteor-installer/install.js [hunk 2]
```diff
@@ -74,7 +71,13 @@ function download() {
   }, cliProgress.Presets.shades_classic);
   downloadProgress.start(100, 0);
 
-  const url = `https://packages.meteor.com/bootstrap-link?arch=os.windows.x86_64&release=${release}`
+  const downloadPlatform = {
```
Comment (filipenevola) [inline npm-packages/meteor-installer/install.js:74]: this is a constant (should be UPPER_CASE) and could be declared in the file scope.

**Review by filipenevola — CHANGES_REQUESTED** [review]
See my recommendations.

**Comment by filipenevola** [comment]
@renanccastro I've got this error on MacOS Intel.

```
npm install -g meteor
/Users/filipe/.nvm/versions/node/v12.14.0/bin/meteor -> /Users/filipe/.nvm/versions/node/v12.14.0/lib/node_modules/meteor/meteor.bat
/Users/filipe/.nvm/versions/node/v12.14.0/bin/meteor-installer -> /Users/filipe/.nvm/versions/node/v12.14.0/lib/node_modules/meteor/cli.js

> meteor@2.3.4 install /Users/filipe/.nvm/versions/node/v12.14.0/lib/node_modules/meteor
> node cli.js install

/Users/filipe/.nvm/versions/node/v12.14.0/lib/node_modules/meteor/config.js:8
  throw new Error('LOCALAPPDATA env var is not set.');
  ^

Error: LOCALAPPDATA env var is not set.
    at Object.<anonymous> (/Users/filipe/.nvm/versions/node/v12.14.0/lib/node_modules/meteor/config.js:8:9)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Function.Module._load (internal/modules/cjs/loader.js:727:14)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/Users/filipe/.nvm/versions/node/v12.14.0/lib/node_modules/meteor/install.js:9:59)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! meteor@2.3.4 install: `node cli.js install`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the meteor@2.3.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/filipe/.npm/_logs/2021-08-18T22_03_18_761Z-debug.log

```

**Comment by filipenevola** [comment]
It seems to be working on MacOS Intel:

```
 meteor-installer git:(feature/installer-linux) ✗ npm link   

> meteor@2.3.5 install /Users/filipe/Documents/meteor/ws/meteor/npm-packages/meteor-installer
> node cli.js install

=> Meteor Downloaded in 27.444s
=> Extracting the tarball, this may take some time
=> Meteor extracted in 25.45s

***************************************

Meteor has been installed!

*You might need to open a new terminal windows to have access to the meteor command.*

To get started fast:

  $ meteor create ~/my_cool_app
  $ cd ~/my_cool_app
  $ meteor

Or see the docs at:

  docs.meteor.com

Deploy and host your app with Cloud:

  www.meteor.com/cloud

***************************************
  
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN meteor@2.3.5 No repository field.

audited 45 packages in 53.304s

2 packages are looking for funding
  run `npm fund` for details

found 2 high severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details
/Users/filipe/.nvm/versions/node/v12.14.0/bin/meteor-installer -> /Users/filipe/.nvm/versions/node/v12.14.0/lib/node_modules/meteor/cli.js
/Users/filipe/.nvm/versions/node/v12.14.0/lib/node_modules/meteor -> /Users/filipe/Documents/meteor/ws/meteor/npm-packages/meteor-installer
```

**Comment by StorytellerCZ** [comment]
On Ubuntu had to go with `sudo` and then still got an error:
```bash
npm ERR! code 1
npm ERR! path /home/storyteller/Web/meteor/meteor/npm-packages/meteor-installer
npm ERR! command failed
npm ERR! command sh -c node cli.js install
npm ERR! => Meteor Downloaded in 41.87s
npm ERR! internal/fs/utils.js:314
npm ERR!     throw err;
npm ERR!     ^
npm ERR! 
npm ERR! Error: EACCES: permission denied, open '/root/.meteor-install-started.txt'
npm ERR!     at Object.openSync (fs.js:498:3)
npm ERR!     at Object.writeFileSync (fs.js:1524:35)
npm ERR!     at b.<anonymous> (/home/storyteller/Web/meteor/meteor/npm-packages/meteor-installer/install.js:102:10)
npm ERR!     at b.emit (events.js:400:28)
npm ERR!     at WriteStream.<anonymous> (/home/storyteller/Web/meteor/meteor/npm-packages/meteor-installer/node_modules/node-downloader-helper/dist/index.js:1:9313)
npm ERR!     at WriteStream.emit (events.js:400:28)
npm ERR!     at emitCloseNT (internal/streams/destroy.js:87:10)
npm ERR!     at processTicksAndRejections (internal/process/task_queues.js:81:21) {
npm ERR!   errno: -13,
npm ERR!   syscall: 'open',
npm ERR!   code: 'EACCES',
npm ERR!   path: '/root/.meteor-install-started.txt'
npm ERR! }

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-08-19T13_04_22_707Z-debug.log
```

BUT it seemed to install and work just fine.

**Comment by renanccastro** [comment]
@StorytellerCZ why you needed sudo? Sudo shouldn't be needed.

**Comment by StorytellerCZ** [comment]
In my case anything global related has to be started with `sudo`, otherwise I get `EACCES` error.

**Comment by renanccastro** [comment]
Got it, on Ubuntu it's common to install nodejs with apt, which in turn will make the global node_modules chowned by root.
I could repro it, and we will need to inform that in these cases if they are running with sudo, they have to pass --unsafe-perm to npm install -g meteor, otherwise, npm won't let the install script run.

Other than that, when running with sudo, the permission of the .meteor folder is wrong, I will also fix it.

**Review by filipenevola — APPROVED** [review]

**Review by StorytellerCZ — APPROVED** [review]

**Comment by ixdi** [comment]
When using npm installer on Linux an error appears:
```
LOCALAPPDATA env var is not set
```
I thought this env was only for Windows users...

**Comment by renanccastro** [comment]
Hello @ixdi . Currently we only support linux/mac in the beta channel of the installer: meteor@2.3.6-beta6.

We will release in the main channel this week.

**Comment by ixdi** [comment]
Thank you @renanccastro for the reply
I was using beta 1 and not beta 6. 
The error continues...
Captura de pantalla de 2021-08-30 15-34-43

**Comment by renanccastro** [comment]
@ixdi in your screenshot seems like the npm install command worked, and the update command failed. NPM is trying to update the package to the main channel, not the beta one. This doesn't seem like an issue with the installer.

**Pull request merged**